test/support/api.ex

```diff
@@ -0,0 +1,90 @@
+defmodule MongoosePush.Support.API do
+  alias HTTPoison.Response
+
+  def sample_notification do
+    %{
+      :service => :apns,
+      :alert => %{
+        :title => "title value",
+        :body => "body value",
+        :click_action => "click.action",
+        :tag => "tag value"
+      }
+    }
+  end
+
+  def post(path, json) do
+    %Response{status_code: status_code, body: body} =
+      HTTPoison.post!(
+        "https://localhost:8443" <> path,
+        Poison.encode!(json),
+        [{"Content-Type", "application/json"}],
+        hackney: [:insecure]
+      )
+
+    %{"details" => details} = Poison.decode!(body)
+    {status_code, details}
+  end
+
+  def mock_apns(json) do
+    {:ok, conn} = get_connection(:apns)
+    payload = Poison.encode!(json)
+
+    headers = headers("POST", "/error-tokens", payload)
+    :h2_client.send_request(conn, headers, payload)
+    get_response(conn)
+    :ok
+  end
+
+  def mock_fcm(json) do
+    {:ok, conn} = get_connection(:fcm)
+    payload = Poison.encode!(json)
+
+    headers = headers("POST", "/mock/error-tokens", payload)
+    :h2_client.send_request(conn, headers, payload)
+    get_response(conn)
+    :ok
+  end
+
+  def reset(:apns) do
+    {:ok, conn} = get_connection(:apns)
+    headers = headers("POST", "/reset")
+    :h2_client.send_request(conn, headers, "")
+    get_response(conn)
+    :ok
+  end
+
+  def reset(:fcm) do
+    {:ok, conn} = get_connection(:fcm)
+    headers = headers("POST", "/mock/reset")
+    :h2_client.send_request(conn, headers, "")
+    get_response(conn)
+    :ok
+  end
+
+  def get_connection(:apns) do
+    :h2_client.start_link(:https, 'localhost', 2197, [])
+  end
+
+  def get_connection(:fcm) do
+    :h2_client.start_link(:https, 'localhost', 4000, [])
+  end
+
+  def headers(method, path, payload \\ "") do
+    [
+      {":method", method},
+      {":authority", "localhost"},
+      {":scheme", "https"},
+      {":path", path},
+      {"content-length", "#{byte_size(payload)}"},
+      {"content-type", "application/json"}
+    ]
+  end
+
+  def get_response(conn) do
+    receive do
+      {:END_STREAM, stream_id} ->
+        :h2_client.get_response(conn, stream_id)
```

Comment:
I know that this code is taken from old test suite, but since we're on it, let's fix most glaring issues - we don't check error code here - we may see random failures in test and not be even aware that mock we're talking to refuses to work. We really should return from this function error code and response body, and match on error code in the functions that actually use it.

Reply:
I saw there was a problem with dialyzer here - if possible I would give it few minutes, and if it doesn't seem like something that is possible to fix, I would just add this file to dialyzer ignore list.

Reply:
Should we match on the possible values returned by `get_response/2` and log when an error occurs?

Reply:
I would suggest to just match and crash. This is an edge case during test setup - we should just know about it before any other side effects occur.